lib/organisms/gravity-form/gravity-form.controller.js

```diff
@@ -127,21 +127,33 @@ function lnOGravityFormController($log, $scope, $element, $attrs, $sce, lnOGravi
   // Init
   //
 
-  $log.debug('lnOGravityFormController -> ', vm, '/ $element: ', $element);
-
-  if (vm.loadFromAPI === true) {
-    lnOGravityFormService.getForm(vm.id)
-      .then(checkLoadFormResponse, checkLoadFormResponse);
+  if (angular.isUndefined(lnOGravityFormService.getConfig())) {
+    var unbindInit = $scope.$on('lnOGravityFormConfigUpdated', init);
+  } else {
+    init();
   }
 
+  function init() {
+    $log.debug('lnOGravityFormController -> ', vm, '/ $element: ', $element);
 
-  vm.form.id = 'gform_' + vm.id;
+    if (angular.isDefined(unbindInit)) {
+      unbindInit();
+    }
 
-  $element.on('submit', _submit);
+    if (vm.loadFromAPI === true) {
```

Comment:
Ok based on your last comment I don't think there is a reason to make a strict comparison here with  `if (vm.loadFromAPI) {` should be enough.